plugins/config_manager.py

```diff
@@ -0,0 +1,241 @@
+#
+# Copyright (C) 2015  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from dnfpluginscore import _, logger
+
+import dnf
+import dnf.cli
+import dnfpluginscore
+import dnfpluginscore.lib
+import os
+import re
+import urlparse
+
+
+class ConfigManager(dnf.Plugin):
+
+    name = 'config-manager'
+
+    def __init__(self, base, cli):
+        super(ConfigManager, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
+        if self.cli is not None:
+            self.cli.register_command(ConfigManagerCommand)
+
+
+class ConfigManagerCommand(dnf.cli.Command):
+
+    aliases = ['config-manager']
+    summary = _('manage dnf configuration options and repositories')
+
+    def __init__(self, cli):
+        super(ConfigManagerCommand, self).__init__(cli)
+        self.opts = None
+        self.parser = None
+
+    def configure(self, args):
+        # setup sack and populate it with enabled repos
+        demands = self.cli.demands
+        demands.available_repos = True
+
+        self.parser = dnfpluginscore.ArgumentParser(self.aliases[0])
+        self.parser.add_argument('repo', nargs='*',
+                    help=_('repo to modify'))
+        self.parser.add_argument('--save',
+                    default=False, action='store_true',
+                    help=_('save the current options (useful with --setopt)'))
+        self.parser.add_argument('--set-enabled',
```

Comment:
Me too :). Unfortunatelly it's not possible because `--enable` is prefix of `--enablerepo` and as such is eaten by dnf when parsing global options and there's no way to forward it to plugin.